### PR TITLE
Only activate when needed

### LINKED ISSUE #86: Use activationHooks to activate the package
After the latest update when I start Atom I get the following error:

```
Failed to spawn command php. Make sure php is installed and on your PATH
```

Yes, PHP is not in my path _right now_, nevertheless the error should only come when I open (lint) a PHP file. I sync my Atom installation across multiple systems and PHP is only installed where I need it.

### PR DESCRIPTION
**Pull request by Arcanemagus**
Utilize the activationHooks in `package.json` to only load this package when a file with the PHP grammar has been opened.

Fixes #86.

### CODE REVIEW

**Comment by Arcanemagus** [comment]
Looks like the specs need to be fixed to force activation.

**Comment by steelbrain** [comment]
@Arcanemagus I think I know what the problem is. On [this line](https://github.com/AtomLinter/linter-php/blob/1aa0d266bd6ab7a2bc2780c983c32b90ed25c0e4/spec/linter-php-spec.js#L7) you need to make it

```js
waitsForPromise(() =>
  atom.packages.activatePackage('language-php').then(() => 
    atom.packages.activatePackage('linter-php')
  )
)
```

File: lib/main.coffee [hunk 1]
```diff
@@ -42,7 +42,8 @@ module.exports =
       lint: (textEditor) =>
         filePath = textEditor.getPath()
         command = @executablePath
-        return [] unless command?
+        if not command?
```
Comment (steelbrain) [inline lib/main.coffee]: `return Promise.resolve([]) unless command?`

File: lib/main.coffee [hunk 2]
```diff
@@ -50,12 +51,13 @@ module.exports =
...
-          regex = /error:\s+(.*?) on line (\d+)/g
-          messages = []
-          while((match = regex.exec(output)) isnt null)
-            messages.push
-              type: "Error"
-              filePath: filePath
-              range: helpers.rangeFromLineNumber(textEditor, match[2] - 1)
-              text: match[1]
-          return messages
+          return new Promise (resolve) ->
```
Comment (steelbrain) [inline lib/main.coffee]: You don't need to return it this way, everything you return from a promise's `.then` it's automatically converted to a promise

File: lib/main.coffee (hunk 1 above)
Comment (Arcanemagus) [inline lib/main.coffee]: Fixed, thanks.

File: lib/main.coffee (hunk 2 above)
Comment (Arcanemagus) [inline lib/main.coffee]: That was part of my attempt to fix the ".then undefined" errors I was seeing, removed.

**Pull request merged**